geopandas/tools/overlay.py

```diff
@@ -300,7 +300,12 @@ def _make_valid(df):
             mask = ~df.geometry.is_valid
             col = df._geometry_column_name
             if make_valid:
+                geom_type = df.geom_type.iloc[0]
                 df.loc[mask, col] = df.loc[mask, col].make_valid()
+                # Extract only the input geometry type, as make_valid may change it
+                if mask.any():
+                    df = _collection_extract(df, geom_type, False)
```

Comment:
```suggestion
                df.loc[mask, col] = df.loc[mask, col].make_valid()
                # Extract only the input geometry type, as make_valid may change it
                if mask.any():
                    df = _collection_extract(df, geom_type='Polygon', keep_geom_type_warning=False)
```
we're in the `if df.geom_type.isin(polys).all()` branch - so I think it's easier to read this if we supply Polygon as the type statically, rather than looking it up. (though I can see the argument this is more consistent with the other branches)